browser/src/control/Control.NavigatorPanel.ts

```diff
@@ -419,42 +419,74 @@ class NavigatorPanel extends SidebarBase {
 		this.builder.build(wrapper, [data], false);
 	}
 
-	override callback(
+	useDefaultCallback(
 		objectType: string,
 		eventType: string,
 		object: any,
 		data: any,
 		builder: JSBuilder,
-	): void {
-		let searchTerm = '';
-		// Update results tab
-		if (object.id === 'navigator-search-button') {
-			searchTerm = (
-				document.getElementById('navigator-search-input') as HTMLInputElement
-			).value;
-			super.callback('edit', 'activate', { id: 'Find' }, searchTerm, builder);
-		} else if (object.id === 'navigator-search') {
-			searchTerm = data;
-			super.callback(
-				objectType,
-				eventType,
-				{ id: 'Find' },
-				searchTerm,
-				builder,
-			);
+	) {
+		super.callback(objectType, eventType, object, data, builder);
+	}
+
+	useSearchCallback(
+		objectType: string,
+		eventType: string,
+		object: any,
+		builder: JSBuilder,
+	) {
+		// Switch to "Results tab" first.
+		if (eventType === 'activate') {
+			const resultsTab = document.getElementById('tab-quick-find');
```

Comment:
could be also:
`notSelectedSearchTab = panel.querySelector('#tab-quick-find:not(".selected")')`